### PR TITLE
Wt 102 implement custom button

### PR DESCRIPTION
**Pull request by k1mka**

### CODE REVIEW

File: lib/presentation/widgets/atoms/main_counter_button_widget.dart [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import 'package:animated_ripple/animated_ripple.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:water_tracker/custom_theme.dart';
+import 'package:water_tracker/generated/assets/assets.gen.dart';
+
+class MainCounterButtonWidget extends StatefulWidget {
```
Comment (AndreyKomarichev10) [inline lib/presentation/widgets/atoms/main_counter_button_widget.dart]: could it be stateless?

File: lib/presentation/widgets/atoms/main_counter_button_widget.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          size: animatedRadius,
+          numberOfRipples: numberOfRipples,
+          duration: duration,
+          color: CustomTheme.mainColor,
+          secondaryColor: CustomTheme.secondaryColor,
+          rippleEffect: RippleEffect.looped,
+          onPressed: widget.onPressed,
+          icon: SvgPicture.asset(
+            Assets.images.waterDrop,
+            width: 40,
```
Comment (AndreyKomarichev10) [inline lib/presentation/widgets/atoms/main_counter_button_widget.dart]: move to const

File: lib/presentation/widgets/molecules/main_screen_counter_widget.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    return Column(
+      children: [
+        spaces,
+        const AssistantWidget(),
+        spaceBetween,
+        RichText(
+          text: TextSpan(
+            children: <TextSpan>[
+              TextSpan(text: LocaleKeys.you_already_got.tr(), style: CustomTheme().gotCupsTextProperty),
+              TextSpan(text: ' $count', style: CustomTheme().countTextProperty),
```
Comment (AndreyKomarichev10) [inline lib/presentation/widgets/molecules/main_screen_counter_widget.dart]: '$count', remove space here

File: lib/presentation/widgets/molecules/main_screen_counter_widget.dart [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+              TextSpan(text: '/$maxCount ', style: CustomTheme().partOfCountTextProperty),
+              TextSpan(text: LocaleKeys.cups.tr(), style: CustomTheme().gotCupsTextProperty),
+            ],
+          ),
+        ),
+        betweenCounters,
+        Align(
+          alignment: Alignment.topLeft,
+          child: Text(
+            '${count * cupWeight}${LocaleKeys.ml.tr()}',
```
Comment (AndreyKomarichev10) [inline lib/presentation/widgets/molecules/main_screen_counter_widget.dart]: you could move it as field

File: lib/water_app.dart [hunk 5]
```diff
@@ -25,7 +25,7 @@ class _WaterAppState extends State<WaterApp> {
         inputDecorationTheme: CustomTheme.inputDecorationTheme,
         textSelectionTheme: CustomTheme.cursorColor,
       ),
-      home: const SignInScreen(),
+      home: const MainScreen(),
```
Comment (AndreyKomarichev10) [inline lib/water_app.dart]: remove it back

File: pubspec.yaml [hunk 6]
```diff
@@ -41,6 +41,12 @@ dependencies:
   flutter_markdown: ^0.6.13
   flutter_keyboard_visibility: ^5.4.0
 
+  animated_ripple:
```
Comment (AndreyKomarichev10) [inline pubspec.yaml:44]: add todo here

File: lib/presentation/screens/main_screen/main_layout.dart [hunk 7]
```diff
@@ -52,40 +44,16 @@ class _MainLayoutState extends State<MainLayout> {
...
-                ),
-                spaceBetweenManAndMl,
-                Align(
-                  alignment: Alignment.topRight,
-                  child: PersonImageWidget.mainWay(),
-                ),
-                spaceBetween,
-                CustomButton(
-                  text: 'ad',
+                MainCounterButtonWidget(
```
Comment (AndreyKomarichev10) [inline lib/presentation/screens/main_screen/main_layout.dart]: mby rename it as MainScreenButtonWidget? and dont forget to rename the file accordingly

**Review by AndreyKomarichev10 — CHANGES_REQUESTED** [review]

File: lib/presentation/widgets/atoms/main_counter_button_widget.dart (hunk 1 above)
Comment (k1mka) [reply lib/presentation/widgets/atoms/main_counter_button_widget.dart]: Yes, if it is stateless i won't be able to create constants

File: lib/presentation/widgets/atoms/main_counter_button_widget.dart (hunk 2 above)
Comment (k1mka) [reply lib/presentation/widgets/atoms/main_counter_button_widget.dart]: 👍🏻

File: lib/presentation/widgets/molecules/main_screen_counter_widget.dart (hunk 3 above)
Comment (k1mka) [reply lib/presentation/widgets/molecules/main_screen_counter_widget.dart]: 👍🏻

File: pubspec.yaml (hunk 6 above)
Comment (k1mka) [reply pubspec.yaml:44]: 👍🏻

File: lib/presentation/widgets/molecules/main_screen_counter_widget.dart (hunk 4 above)
Comment (k1mka) [reply lib/presentation/widgets/molecules/main_screen_counter_widget.dart]: The instance member 'count' can't be accessed in an initializer.

File: lib/presentation/screens/main_screen/main_layout.dart [hunk 8]
```diff
@@ -52,40 +44,16 @@ class _MainLayoutState extends State<MainLayout> {
...
                   onPressed: () => context.read<MainScreenBloc>().add(SaveCounterEvent()),
-                  buttonColor: Colors.red,
-                  textButtonColor: Colors.red,
+                ),
+                MainScreenButtonWidget(
```
Comment (zddarova) [inline lib/presentation/screens/main_screen/main_layout.dart]: rename it back to MainScreenCounterWidget

**Review by zddarova — CHANGES_REQUESTED** [review]

File: lib/presentation/widgets/molecules/main_counter_button_widget.dart [hunk 9]
```diff
@@ -0,0 +1,55 @@
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:water_tracker/custom_theme.dart';
+import 'package:water_tracker/generated/locale_keys.g.dart';
+import 'package:water_tracker/presentation/widgets/atoms/person_image_widget.dart';
+import 'package:water_tracker/presentation/widgets/molecules/assistant_widget.dart';
+
+class MainScreenCounterWidget extends StatelessWidget {
+  const MainScreenCounterWidget({super.key, required this.count, required this.maxCount, required this.cupWeight})
+      : counterCupWeight = count * cupWeight;
```
Comment (zddarova) [inline lib/presentation/widgets/molecules/main_counter_button_widget.dart]: rename pls. For me, it is unclear that counterCupWeight means max weight. I suggest maxWaterWeigth or waterLimit

File: lib/presentation/screens/main_screen/main_layout.dart [hunk 10]
```diff
@@ -21,12 +18,7 @@ class MainLayout extends StatefulWidget {
 class _MainLayoutState extends State<MainLayout> {
   static const cupsMaxCount = MainScreenBloc.maxCupCount;
   static const singleCupWeight = MainScreenBloc.singleCupWeight;
-
   static const paddingHorizontal = EdgeInsets.symmetric(horizontal: 28.0);
-  static const spaces = SizedBox(height: 24);
```
Comment (zddarova) [inline lib/presentation/screens/main_screen/main_layout.dart:26]: why deleted this?

**Review by zddarova — APPROVED** [review]

**Pull request merged**